docs/apis/subsystems/group/index.md

```diff
@@ -134,6 +152,29 @@ $url = new moodle_url('/mod/forum/view.php', ['id' => $cm->id]);
 groups_print_activity_menu($cm, $url);
 ```
 
+### How to get just the groups that the current user can see
+
+The following example will check whether the current user has permission to see hidden groups on a course, and **if they do not**, will apply additional conditions to a query to restrict the results to just those groups they should see. Note that a query like this must join on `group_members` as group visibility is dependant on the user's own memberships.
+
+```php
+$courseid = required_param('courseid', PARAM_INT)';
+$sql = "SELECT g.idnumber, gm.*
+          FROM {groups} g
+          JOIN {groups_members} gm ON gm.groupid = g.id
+         WHERE courseid = ?";
+
+$params = [$courseid];
+
+if (!has_capability('moodle/course:viewhiddengroups', context_course::instance($courseid)) {
+    // Apply visibility restrictions.
+    list($visibilitywhere, $visibilityparams) = \core_group\visibility::sql_group_visibility_where($userid);
```

Comment:
When it comes to code examples, it's good to keep them fairly narrow so that they don't overflow on mobile too much where possible.

I'd also suggest swaping from the `list` to the `[]` syntax.

```suggestion
    [
        $visibilitywhere,
        $visibilityparams,
     ] = \core_group\visibility::sql_group_visibility_where($userid);
```